Modify Settings and User-Name

gauravbhardwaj7 opened the pull request:
This PR has following changes-
1.) Modified SettingsView and added dark mode functionality and User token input.
2.) Made username dynamic .

gauravbhardwaj7 committed:
added sheet settings

gauravbhardwaj7 committed:
Added Dark mode

gauravbhardwaj7 committed:
Made username dynamic

gauravbhardwaj7 committed:
User token

Created text field for user to enter user token

gauravbhardwaj7 committed:
Updated toolbar

In the toolbar, improved placement of icons and added Save button in Settings View.

akshaaatt commented on Listenbrainz/UI/Components/SettingsView.swift:
```diff
@@ -0,0 +1,54 @@
+struct SettingsView: View {
+  @AppStorage("isDarkMode") private var isDarkMode = false
+
+  @State private var userToken:String = ""
```
Use AppStorage here as well

gauravbhardwaj7 replied:
done

akshaaatt commented on Listenbrainz/UI/Screens/SongDetailView.swift:
```diff
@@ -8,38 +8,79 @@
+  var body: some View {
+    VStack {
+      if !listensFetched {
+        TextField("Enter Username", text: $userNameInput)
```
Please remove this from here and just add a username field in the settings page directly

gauravbhardwaj7 replied:
done

akshaaatt commented on Listenbrainz/UI/Components/SettingsView.swift:
```diff
@@ -0,0 +1,54 @@
+  @AppStorage("isDarkMode") private var isDarkMode = false
+
+  @State private var userToken:String = ""
+  @Environment(\.presentationMode) var presentationMode
```
Have a look at the dismiss Environment. presentationMode has been deprecated so we can switch

gauravbhardwaj7 replied:
done

akshaaatt reviewed (changes requested):
Thanks a lot for the work here!
Just a few changes then we should be good to proceed.

gauravbhardwaj7 committed:
Updated API response

gauravbhardwaj7 committed:
Added User-Name in SettingsView

In this commit, User can enter the username in the settings view and can get listens accordingly

gauravbhardwaj7 committed:
Used .dismiss instead of .presentationMode

gauravbhardwaj7 committed:
Added BrainzPlayerMetadata
